src/main/java/com/workshop/demo/controllers/RestaurantController.java

```diff
@@ -42,8 +43,7 @@ public Integer getScore(
         return score;
     }
 
-    @PostMapping
-    @PreAuthorize("hasRole('USER')")
+    @PostMapping("/add")
     public Restaurant addRestaurant(@Valid @RequestBody RestaurantRequest restaurantRequest,
             @CurrentUser User user) {
```

Comment:
Change this to `@CurrentUser UserPrincipal user`. 
Ref: option 3 in https://www.baeldung.com/get-user-in-spring-security.